plugins/search-backend-module-pg/src/PgSearchEngine/PgSearchEngineIndexer.ts

```diff
@@ -92,4 +92,32 @@ export class PgSearchEngineIndexer extends BatchSearchEngineIndexer {
       throw e;
     }
   }
+
+  /**
+   * Custom handler covering the case where an error occurred somewhere else in
+   * the indexing pipeline (e.g. a collator or decorator). In such cases, the
+   * finalize method is not called, which leaves a dangling transaction and
+   * therefore an open connection to PG. This handler ensures we close the
+   * transaction and associated connection.
+   *
+   * todo(@backstage/discoverability-maintainers): Consider introducing a more
+   * formal mechanism for handling such errors in BatchSearchEngineIndexer and
+   * replacing this method with it. See: #17291
+   *
+   * @internal
+   */
+  async _destroy(error: Error | null, done: (error?: Error | null) => void) {
+    // Ignore situations where there was no error.
+    if (!error) {
+      done();
+      return;
+    }
+
+    try {
+      this.tx!.rollback(error);
```

Comment:
We should probably `await` this, right?

Also, how about using `isCompleted` (TIL, cf. [knex transaction docs](https://knexjs.org/guide/transactions.html)):

```typescript
if (!this.tx!.isCompleted()) {
  await this.tx!.rollback();
}
```

Reply:
Oooh, nice.  I wish the knex TS had better hints--it's unclear without looking at the docs that rollback returns a promise.

I notice all of the other rollback calls are also not awaited in this class.  Think it's worth adding that in elsewhere too?

Reply:
For the sake of focus/simplicity, I avoided touching other parts of the class and went with this suggestion verbatim.  If there are problems that arise where rollback itself fails, I'd hope we'd get report of that, and we could solve that all in one go.